File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/command/memento_solution/turn_left_command.cpp
// turn_left_command.cpp

#include "driver.h"
#include "turn_left_command.h"

turn_left_command::turn_left_command(driver * a_driver)
: simple_driver_command(a_driver)
{
}

void turn_left_command::execute()
{
	simple_driver_command::execute();
	get_driver()->turn_left();
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/command/solution/TurnLeftCommand.java
// TurnLeftCommand.java

public class TurnLeftCommand extends SimpleDriverCommand
{
	public TurnLeftCommand(Driver aDriver)
	{
		super(aDriver);
	}

	public void execute()
	{
		getDriver().turnLeft();
	}

	public void undo()
	{
		getDriver().undoTurnLeft();
	}
}
